code/modules/events/spider_terror.dm

```diff
@@ -44,11 +43,18 @@
 			// Strongest, only used during highpop.
 			spider_type = /mob/living/simple_animal/hostile/poison/terror_spider/queen
 			spawncount = 1
-	while(spawncount && length(vents))
-		var/obj/vent = pick_n_take(vents)
-		new spider_type(vent.loc)
-		spawncount--
-		successSpawn = TRUE
+	spawn() // Just like in xeno and demon events, this spawn() is necessary. Do not remove.
```

Comment:
Why is this necessary? What else needs to run before the code below?

Reply:
Without it, the whole thing breaks. The event loops and it generates incorrect popups.